src/libraries/System.Globalization/tests/CultureInfo/GetCultureInfo.cs

```diff
@@ -32,12 +32,15 @@ public static IEnumerable<object[]> GetCultureInfoTestData()
 
             if (PlatformDetection.IsIcuGlobalization)
             {
-                yield return new object[] { "x\u0000X-Yy", "x" }; // Null byte
+                if (PlatformDetection.IsNotWindows)
```

Comment:
Is there a test anywhere for this scenario in the PR description?

> In the file IcuLocaleData.cs, there was a bug that caused not finding the proper culture name mapping to our data because the culture name containing characters like underscore character _.

Reply:
Yes, we have the test https://github.com/dotnet/runtime/blob/c6742603fc6c6f8d503ff8e000aa431c394df87d/src/libraries/System.Globalization/tests/CultureInfo/CultureInfoAll.cs#L661. It is funny that someone inside this test added the condition https://github.com/dotnet/runtime/blob/c6742603fc6c6f8d503ff8e000aa431c394df87d/src/libraries/System.Globalization/tests/CultureInfo/CultureInfoAll.cs#L670

```C#
                if (ci.LCID == 0x1000)
                {
                    // Unsupported culture.
                    return;
                }
```

which was hiding the problem I fixed. I have removed this condition now and now the scenario is covered by this test.

Reply:
It looks like that check was added for WASM (#47301). We should make sure the WASM tests still pass.

Reply:
There is outer condition https://github.com/dotnet/runtime/blob/c6742603fc6c6f8d503ff8e000aa431c394df87d/src/libraries/System.Globalization/tests/CultureInfo/CultureInfoAll.cs#L663which I think excluding WASM runs.